overlays/python-inconsistent-interpreters.nix

```diff
@@ -0,0 +1,39 @@
+{ builtAttrs
+, packageSet
+, namePositions
+}@attrs:
+
+final: prev:
+let
+  inherit (prev) lib;
+  inherit (import ../lib { inherit lib; }) checkBuildPytonPackageFor;
+
+  checkDerivation = python: drv:
+      let
+            propagatedBuildInputs = lib.attrByPath ["propagatedBuildInputs"] [] drv;
+            checkInputs = lib.attrByPath ["checkInputs"] [] drv;
+            runtimeInputs = checkInputs ++ propagatedBuildInputs;
+            inputPythonInterpreters = map (p: p.pythonModule) (builtins.filter (p: p ? pythonModule) runtimeInputs);
+            allPythonInterpreters = lib.unique (inputPythonInterpreters ++ [python]);
+        in
+    lib.singleton {
+        name = "python-inconsistent-interpreters";
+        cond = lib.length allPythonInterpreters > 1;        
+        msg = let
+            allPythonNames = map (p: p.name) allPythonInterpreters;
+            sources = ["buildPythonPackage"] ++ 
+                (if (lib.length propagatedBuildInputs) > 0 then ["propagatedBuildInputs"] else []) ++
+                (if (lib.length checkInputs) > 0 then ["checkInputs"] else []);            
+        in            
+        ''
+          Between ${lib.concatStringsSep " and " sources}, this derivation seems
+          to be simultaneously trying to use packages from multiple different python package sets.
+          Mixing package sets like this is not supported.
+
+          Detected pythons:
+
+            ${lib.concatStringsSep "\n  " allPythonNames}
```

Comment:
It might be nice to name the offending packages. Though most of the time, they would probably be listed included using `python*Packages` in the file so maybe it is unnecessary.